crates/wasmtime/src/externals.rs

```diff
@@ -598,6 +598,29 @@ impl Table {
     }
 }
 
+/// An error that can occur when working with [`Memory`].
+#[derive(Debug, PartialEq)]
+#[non_exhaustive]
+pub enum MemoryError {
+    /// Out of bounds memory read/write.
+    OutOfBounds,
+    /// The memory is already locked by another reader or writer.
+    // Note: currently unused.
+    Locked,
```

Comment:
FWIW I don't think that we'll ever need this error because we'll basically be implementing `memory.copy` semantics which would never need an error for this.

Reply:
In that case ... should the error just be a struct? 

I don't see any other relevant error conditions.

Reply:
That's true yeah, and I think we could go ahead and have it just be an opaque structure with a singular private field so we can add more to it later if we want.

Reply:
Done.